### PR TITLE
Fix: Fixed type errors revealed by Cypress.

### PR DESCRIPTION
**Pull request by James-Nitron**
Type issues: 

Map.svelte:  
`container needs to be either a string or HTMLElement. `

Fix: Wait for the container to be defined before generating the map.

Tileset.svelte:
`"Cannot read properties of undefined (reading 'getSource')"`

Fix: map.isStyleLoaded(id) was causing the error. A Github discussion suggested using `map.on(style.loaded)` which is an undocumented Mapbox lifecycle event. Using the above stops these errors from generating and after comparing to https://ons-data-map.netlify.app/ looks as if everything still works as expected and seems like there might be a slight speed increase with styles loading.

ONSAutosuggest.svelte
`Uncaught TypeError: Cannot read property 'innerText' of null`

Fix: Check input exists before attempting to extract innerText.

### CODE REVIEW

**Comment by designorant** [comment]
Nice one!

Would you mind linking the discussion to that undocumented lifecycle method?

**Comment by James-Nitron** [comment]
> Nice one!
> 
> Would you mind linking the discussion to that undocumented lifecycle method?

Good shout, link is https://github.com/mapbox/mapbox-gl-js/issues/2268

**Comment by PeteMontgomery** [comment]
Hmm, using an undocumented API sounds quite a drastic solution.

**Comment by PeteMontgomery** [comment]
My thoughts are it points to something else being wrong, but I’m happy for now if it gets marked with a big warning comment and a link to @mourner’s comment. @designorant

File: src/ui/map/TileSet.svelte [hunk 1]
```diff
@@ -39,15 +39,9 @@
...
-  function isMapLoaded() {
-    if (map.isStyleLoaded(id)) {
-      addSource();
-    } else {
-      setTimeout(() => {
-        isMapLoaded();
-      }, 100);
-    }
-  }
+  map.on("style.load", () => {
```
Comment (designorant) [inline src/ui/map/TileSet.svelte:42]: Let's stick a comment in there as per Pete's suggestion and ship it.

Comment (James-Nitron) [reply src/ui/map/TileSet.svelte:42]: Wicked, I'll sort that now.

**Review by designorant — APPROVED** [review]
🚀

**Pull request merged**